imagersite/imager_profile/tests.py

```diff
@@ -60,3 +78,81 @@ def test_delete_profile_deletes_user(self):
         self.assertEqual(len(ImagerProfile.objects.all()), 1)
         user.profile.delete()
         self.assertEqual(len(User.objects.all()), 0)
+
+
+class ProfileViewTestCase(TestCase):
+    def setUp(self):
+        user = UserFactory(username='userbob', email='bob@example.com')
+        user.set_password('secret')
+        user.save()
+        user.profile.camera = 'Canon'
+        user.profile.address = '123 Bob Drive'
+        user.profile.website = 'www.bob.com'
+        user.profile.photography_type = 'portraits'
+        PhotoFactory(title='Bobs Photo', user=user)
+        AlbumFactory(title='Bobs Album', user=user)
+        user.profile.save()
+
+        user2 = UserFactory(username='usereve', email='eve@example.com')
+        user2.set_password('secret')
+        user2.save()
+
+    def test_view_profile(self):
+        c = Client()
+        c.login(username='userbob', password='secret')
+        response = c.get('/profile/')
+        self.assertIn('Your camera is a Canon', response.content)
+        self.assertIn('You have <strong>1</strong> albums', response.content)
+        self.assertIn('You have a total of <strong>1</strong> photos',
+                      response.content)
+
+    def test_eve_doesnt_see_bob(self):
+        c = Client()
+        c.login(username='usereve', password='secret')
+        response = c.get('/profile/')
+        self.assertIn('You have <strong>0</strong> albums', response.content)
+
+    def test_unauthenticated_redirect(self):
+        c = Client()
+        response = c.get('/profile/', follow=True)
+        self.assertEqual(len(response.redirect_chain), 1)
+        self.assertIn('form method="post" action="/login/"',
+                      response.content)
+
+
+def ProfileEditTestCase(TestCase):
+    def setUp(self):
+        user = UserFactory(username='userbob', email='bob@example.com')
+        user.set_password('secret')
+        user.save()
+        user.profile.camera = 'Canon'
+        user.profile.address = '123 Bob Drive'
+        user.profile.website = 'www.bob.com'
+        user.profile.photography_type = 'portraits'
+        user.profile.save()
+
+    def test_profile_edit_view(self):
+        c = Client()
+        c.login(username='userbob', password='secret')
+        user = User.objects.all()[0]
+        response = c.get('/profile/edit/')
```

Comment:
Took me until now to notice this, but it's often a good idea to use the python equivalent of Django's `url` template tag, which is `reverse` in order to avoid hard-coding your urls into test files like this.